### PR TITLE
Bw 264

### PR DESCRIPTION
**Pull request by mehrlerKinsol**
Adds equal error rate metric and outputs metrics to a JSON file to work with DVC

### CODE REVIEW

File: Dockerfile
Comment (mehrlerKinsol) [inline Dockerfile:36]: Not sure why these are showing up as changed. Torch 1.9 should be in master already

File: yolo/utils/metrics.py
Comment (kweston) [inline yolo/utils/metrics.py]: This isn't returning the actual threshold. I think you need to pass the thresholds in as a 3rd function argument and return the one that triggers the change.

File: yolo/utils/metrics.py
Comment (kweston) [inline yolo/utils/metrics.py]: Not sure why you are interpolating here, can't you just read in the curves corresponding to the 0.5 IoU threshold directly?

File: test.py
Comment (kweston) [inline test.py]: I think the file Path object should be passed in as an optional CLI arg. Only write the JSON file if the filename is present

File: test.py
Comment (kweston) [inline test.py]: I prefer to use the pathlib.Path API where possible.
`filename.parent.mkdir(exist_ok=True, parents=True)`

File: yolo/utils/metrics.py
Comment (kweston) [inline yolo/utils/metrics.py]: I think this should be only called if a `computer_eer` function arg is present.

**Review by kweston — CHANGES_REQUESTED** [review]

File: yolo/utils/metrics.py
Comment (mehrlerKinsol) [reply yolo/utils/metrics.py]: The thresholds in the conf array don't quite map to the right threshold so some kind of mapping is needed. I found a bit cleaner way to do it though

File: yolo/utils/metrics.py
Comment (mehrlerKinsol) [reply yolo/utils/metrics.py]: If this argument is false and a filename for metrics is given what should it output? Just 0s?

Comment (kweston) [reply yolo/utils/metrics.py]: 0s or NaNs I'm fine with either

**Review by kweston — APPROVED** [review]
Remember to squash commits before merging. Maybe we should hold off on merging this until the torch1.9 issues are sorted out.

Also, we usually format our PRs as
BW-264 <short description of ticket>
- <first change>
- <second change>
- ...

**Pull request merged**
